### PR TITLE
add new plugin proxy-port-forward

### PR DESCRIPTION
**Pull request by kvaps**
A new plugin from `kubectl node-shell` developer (me).
Might be useful for local development, when you need to proxy multiple services including external ones.

### CODE REVIEW

**Comment by ahmetb** [comment]
:robot: _Beep beep! I’m a robot speaking on behalf of @ahmetb._ :robot:

-----

Thanks for submitting your kubectl plugin to Krew!
One of the krew-index maintainers will review it soon. Note that the reviews for new plugin submissions may take a few days.

In the meanwhile, here are a few tips to make your plugin manifest better:
* Make sure your plugin follows the [best practices](https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/).
* Eliminate redundant wording form `shortDescription` (it should be max 50 characters).
* Try to word wrap your `description` to 80-character lines (no usage examples, please).

Thanks for your patience!
/kind new-plugin

File: plugins/proxy-port-forward.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  shortDescription: Proxy to remote host via Kubernetes
+  description: |
+    Start socat with proxy to remote host in Kubernetes and forward one or more local ports to it.
+
+    Usage:
+        $ kubectl proxy-port-forward <local_port>:<remote_host>:<remote_port> [<local_port>:<remote_host>:<remote_port>]
+
+    Example:
+        $ kubectl proxy-port-forward 8443:google.com:443 6443:kubernetes.default.svc:443
+        $ curl -k https://localhost:6443
```
Comment (ahmetb) [inline plugins/proxy-port-forward.yaml:17]: let's remove these from here. they get outdated. this should be in -h/--help of the plugin.

File: plugins/proxy-port-forward.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: proxy-port-forward
+spec:
+  version: v1.0.0
+  homepage: https://github.com/kvaps/kubectl-proxy-port-forward
+  shortDescription: Proxy to remote host via Kubernetes
+  description: |
+    Start socat with proxy to remote host in Kubernetes and forward one or more local ports to it.
```
Comment (ahmetb) [inline plugins/proxy-port-forward.yaml:10]: Q: Are there no `caveats` to this tool at all, like maybe the pod requires `socat`? 

In the code I see things like `kubectl delete pod`, which I would not expect this plugin to do. I think this field is a good opportunity to make it clear to users how it works even more.

Comment (kvaps) [reply plugins/proxy-port-forward.yaml:10]: The approach is similar to `node-shell` plugin.

A plugin spawns a new pod with socat to proxy all requests to specified remote hosts, which are available only from Kubernetes cluster.
After usage the pod obviously should be removed.

Comment (ahmetb) [reply plugins/proxy-port-forward.yaml:10]: I think you explained it really well here. This can be made clear in the description too.

**Comment by ahmetb** [comment]
Pinging to see if you're interested in updating.

**Comment by ahmetb** [comment]
Ping.

**Comment by kvaps** [comment]
Let's close it for now, since I have no time to maintain this project anymore

/close

**Pull request closed without merging**